crates/fluvio-compression/src/lib.rs

```diff
@@ -58,10 +58,14 @@ impl FromStr for Compression {
 
 impl Compression {
     /// Compress the given data, returning the compressed data
-    pub fn compress(&self, src: &[u8]) -> Result<Vec<u8>, CompressionError> {
+    pub fn compress(
```

Comment:
I think that if we wrap CompressionLevel internally in Compression type, we would not need to add level to `compress()`.

Maybe:
```rust
pub struct Compression {
   kind: CompressionAlgorithm,
   level: Option<CompressionLevel>
 }
// or:
 pub enum Compression {
   None,
   Gzip(CompressionLevel),
   Snappy,
    Lz4
 }
 ```
 
 With that we won't need to pass compression_lvel to Batch structs. We would have to change the encoder/decoder implementation to only consider the  compression algorithm

Reply:
With the second approach we can specialize `CompressionLevel` to correspond exactly to what Gzip offers, and we can also generalize the parameter to more than just compression level. LZ4, for example, has some other compression options, like choosing the max block size and whether the compression dictionary is carried over to the next block.

Reply:
A problem I see with non-unit enum variants is that you can't use the explicit discriminant with them.